app/controllers/social_previews_controller.rb

```diff
@@ -5,7 +5,7 @@ class SocialPreviewsController < ApplicationController
 
   def article
     @article = Article.find(params[:id])
-    @tag_badges = Badge.where(id: Tag.where(name: @article.decorate.cached_tag_list_array).pluck(:badge_id))
+    @tag_badges = Badge.where(id: Tag.where(name: @article.decorate.cached_tag_list_array).select(:badge_id))
```

Comment:
Rails is smart enough to build a SQL subquery when given an ActiveRecord relation in a `.where()`. Unfortunately `.pluck` is an eager method, which results in 2 queries (inner pluck, plus external where) instead of 1 SQL query.

Thus `.select()` is preferable here over `.pluck()` (and we also save some memory in case of huge arrays)

See https://github.com/rubocop-hq/rubocop-rails/issues/246

Reply:
Nice optimization! 😃